test/ably/restchannelpublish_test.py

```diff
@@ -231,6 +234,20 @@ def test_message_attr(self):
         self.assertEqual(message.client_id, 'client_id')
         self.assertIsInstance(message.timestamp, int)
 
+    def test_token_is_bound_to_options_client_id_after_publish(self):
+        # null before publish
+        self.assertIsNone(self.ably_with_client_id.auth.token_details)
+
+        # created after message publish and will have client_id
+        channel = self.ably_with_client_id.channels[
+            self.protocol_channel_name('persisted:restricted_to_client_id')]
+        channel.publish(name='publish', data='test')
```

Comment:
If you're publishing, can we not check that the message published when retrieved via history has the intended client_id?